docs/design/communication_library.md

```diff
@@ -0,0 +1,23 @@
+## Communication Library
+
+We believe there is no need to re-implement a communication library compatible with data structures differing from this dl framework to another. Therefore, FTLib will import a user-specific communication library during initialization. 
+
+Some deep learning frameworks wrap communication libraries, such like NCCL or Gloo, and make it compatible with its own data structure. Such wrapped communication libraries can be adopted with less effort to be compatible with deep learning frameworks.
+
+Since we are working with ElasticDL, some low-level APIS are exposed.
```

Comment:
Maybe rephrase to "There are also deep learning frameworks that require highly customized usage of the collective communication APIs. In this case, we also expose several low-level APIs to facilitate that."?

Reply:
touché